src/ORM/AbstractORMPersistenceStrategy.php

```diff
@@ -152,18 +152,29 @@ private function dropAndResetDatabase(Application $application): void
 
     private function createSchema(Application $application): void
     {
-        if (self::RESET_MODE_MIGRATE === $this->config['reset']['mode']) {
+        if (self::RESET_MODE_SCHEMA === $this->config['reset']['mode']) {
+            foreach ($this->managers() as $manager) {
+                self::runCommand($application, 'doctrine:schema:update', [
+                    '--em' => $manager,
+                    '--force' => true,
+                ]);
+            }
+
+            return;
+        }
+
+        if (!$migrationsConfigurations = $this->config['reset']['migrations']['configurations']) {
             self::runCommand($application, 'doctrine:migrations:migrate', [
                 '--no-interaction' => true,
             ]);
 
             return;
         }
 
-        foreach ($this->managers() as $manager) {
-            self::runCommand($application, 'doctrine:schema:update', [
-                '--em' => $manager,
-                '--force' => true,
+        foreach ($migrationsConfigurations as $migrationsConfiguration) {
```

Comment:
What of no migration is provided? I'm surprized that the tests still pass 🤔

Reply:
In that case `doctrine:migrations:migrate` is run as before without its `--configuration` option. I changed the conditions order to simplify the code so now it’s done in line 167.